planner/core/plan_cost_ver2.go

```diff
@@ -38,6 +38,9 @@ func GetPlanCost(p PhysicalPlan, taskType property.TaskType, option *PlanCostOpt
 func getPlanCost(p PhysicalPlan, taskType property.TaskType, option *PlanCostOption) (float64, error) {
 	if p.SCtx().GetSessionVars().CostModelVersion == modelVer2 {
 		planCost, err := p.getPlanCostVer2(taskType, option)
+		if traceCost(option) {
+			genPlanCostTrace(p, planCost, taskType, option)
+		}
```

Comment:
Everything else looks good to me, except for `plan_cost_trace.go`, which seems duplicated with `plan_cost_ver2.go`. Therefore, I don't recommend merging this file into the Master Branch. 
How about merging all other parts except `plan_cost_trace.go`? 
We can leave a hook function in place so that you can initialize it on your Branch and use this trace feature:

For example, on Mater, we leave a hook function:
```
var genPlanCostTrace func (...)
...
func getPlanCost(...) {
    ....
    if traceCost(option) && genPlanCostTrace != nil {
        genPlanCostTrace(p, planCost, taskType, option)
    }
}
...
```

And on your Branch, initialize this hook function through `init()`:
```
func init() {
    genPlanCostTrace = yourGenPlanCostTrace
}
```

Reply:
PTAL @pingandb